### ISSUE TITLE
Search this area of the map button: behavior and refactor

### ISSUE THREAD

**Issue by mmiddaugh** [department-of-veterans-affairs/va.gov-team#21958]
## Issue Description
The appearance of the "Search this area of the map" button is no longer tied to an interaction with the map. This has disrupted focus management and announcement to screenreaders, causing accessibility issues.  **This issue deals with the behavior and refactoring - accessibility will be addressed in #21256** 

## Steps to reproduce

1. Populate _City, state or postal code_ box with 59860
2. Select VA health as _Facility type_
3. Click search
4. Observe map and list of search results.
5. Observe the _Search this area of the map_ button appears before any interaction with the map has occurred. 

# Tasks
- [ ] Return behavior to original functionality: button is available after interacting with the map. 
- [ ] Convert the Search this area of the map button to a React component (at least 5 pts worth of work)

## Acceptance Criteria
- [ ] _Search this area of the map_ button will not display unless the Veteran has interacted with the map by panning or zooming. 
- [ ] _Search this area of the map_ button receives focus when displayed. 
- [ ] The _Search this area of the map_ button does not execute a search when the Facility Type is not set, or Specialty Type is not set for CCP search. Button is not actionable and uses [disabled color scheme](https://design.va.gov/components/buttons#disabled-buttons)
- [ ] Button is labeled _Search this area of the map_ if the search radius is under 500 miles 
- [ ] Button is labeled _Zoom in to search_ if the search radius is over 500 miles, is not actionable and uses [disabled color scheme](https://design.va.gov/components/buttons#disabled-buttons)
- [ ] Tests exist to cover the above AC

**Comment by mpelzsherman** [comment]
Moving to blocked based on conflicting feedback from this [Slack thread](https://dsva.slack.com/archives/C0NGDDXME/p1617676715140000).
